force_torque_sensor_broadcaster/src/force_torque_sensor_broadcaster.cpp

```diff
@@ -0,0 +1,175 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * Author: Subhas Das, Denis Stogl
+ */
+
+#include "force_torque_sensor_broadcaster/force_torque_sensor_broadcaster.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace force_torque_sensor_broadcaster
+{
+
+ForceTorqueSensorBroadcaster::ForceTorqueSensorBroadcaster()
+: controller_interface::ControllerInterface()
+{}
+
+controller_interface::return_type
+ForceTorqueSensorBroadcaster::init(const std::string & controller_name)
+{
+  auto ret = ControllerInterface::init(controller_name);
+  if (ret != controller_interface::return_type::OK) {
+    return ret;
+  }
+
+  try {
+    auto node = get_node();
+    node->declare_parameter<std::string>("sensor_name", "");
+    node->declare_parameter<std::string>("interface_names.force.x", "");
+    node->declare_parameter<std::string>("interface_names.force.y", "");
+    node->declare_parameter<std::string>("interface_names.force.z", "");
+    node->declare_parameter<std::string>("interface_names.torque.x", "");
+    node->declare_parameter<std::string>("interface_names.torque.y", "");
+    node->declare_parameter<std::string>("interface_names.torque.z", "");
+    node->declare_parameter<std::string>("frame_id", "");
+  } catch (const std::exception & e) {
+    fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
+    return controller_interface::return_type::ERROR;
+  }
+
+  return controller_interface::return_type::OK;
+}
+
+CallbackReturn ForceTorqueSensorBroadcaster::on_configure(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  sensor_name_ = node_->get_parameter("sensor_name").as_string();
+  interface_names_.resize(6, "");
+  interface_names_[0] = node_->get_parameter("interface_names.force.x").as_string();
+  interface_names_[1] = node_->get_parameter("interface_names.force.y").as_string();
+  interface_names_[2] = node_->get_parameter("interface_names.force.z").as_string();
+  interface_names_[3] = node_->get_parameter("interface_names.torque.x").as_string();
+  interface_names_[4] = node_->get_parameter("interface_names.torque.y").as_string();
+  interface_names_[5] = node_->get_parameter("interface_names.torque.z").as_string();
+
+  if (sensor_name_.empty() &&
+    std::count(interface_names_.begin(), interface_names_.end(), "") == 6)
+  {
+    RCLCPP_ERROR(
+      get_node()->get_logger(), "'sensor_name' or at least one "
+      "'interface_names.[force|torque].[x|y|z]' parameter has to be specified.");
+    return CallbackReturn::ERROR;
+  }
+
+  if (!sensor_name_.empty() &&
+    std::count(interface_names_.begin(), interface_names_.end(), "") != 6)
+  {
```

Comment:
let's save re-counting those `""`-s and make the code a bit more readable
```suggestion

  const bool no_interface_names_defined = std::count(interface_names_.begin(), interface_names_.end(), "") == 6;

  if (sensor_name_.empty() && no_interface_names_defined)
  {
    RCLCPP_ERROR(
      get_node()->get_logger(), "'sensor_name' or at least one "
      "'interface_names.[force|torque].[x|y|z]' parameter has to be specified.");
    return CallbackReturn::ERROR;
  }

  if (!sensor_name_.empty() && !no_interface_names_defined)
  {
```